### DISCUSSION TITLE
How to use trigger() for section-level validation?

### DISCUSSION THREAD

**Discussion by Aryaman73** [react-hook-form/react-hook-form#6057] [Q&A] [answered]
We're using dot-notation while registering our inputs to divide them into sections, e.g. `{...register("section1.last-name")}`. The output of `watch()` returns something like this:

`{... "section1":{"last-name":"Singh", "first-name":"Aryaman", "middle-initial":"M"} ... }`

We want to use `trigger()` to validate each section, but `trigger("section1")` doesn't work; it only seems to work for `trigger(section1.last-name)` (i.e only for a particular input)

Is there a way to get it to work with sections? From reading the documentation it seems that `trigger()` was meant for single-inputs only, so is there another way to get this functionality?

(Just for clarity, this is not a "multi-step" form as the examples state; all inputs are within in the same `<form>` tags and we aren't using state machine/redux)

**Comment by bluebill1049** [comment]
you can upvote on this feature request: https://github.com/react-hook-form/react-hook-form/projects/1

**Comment by Aryaman73** [comment]
Thanks for the reply @bluebill1049 , hopefully this feature gets added soon!

-----
One workaround that I discovered might be to do something like this:
`  const fieldsInMySection = Object.keys(getValues()[mySection])`

Which gets all the keys in the given section, i.e. `["first-name", "last-name" ...]`

And then you can just run a forEach:

```
fieldsInMySection.forEach((id: string) => {
          trigger(`${mySection}.${id}`);
});
```

This seems to be working for now, but I'd be open to any suggestions or comments.

**Reply by bluebill1049** [reply]
Confirmed to be the next minor release.

**Reply by Aryaman73** [reply]
I can see this was merged in: https://github.com/react-hook-form/react-hook-form/pull/5958

**Reply by bluebill1049** [reply]
you can try the `next` release 7.13.0-next.0

**Comment by Aryaman73** [comment, marked as answer]
React Hook Forms now supports passing both **field array** and **object fields** into trigger(). As of this comment, it is only in the .next release but it should make it to the release branch soon.

 Thanks a lot, this really makes validation easier! 🥳 
 
 See: https://github.com/react-hook-form/react-hook-form/pull/5958
